quintoandar_kafka/kafka_consumer.py

```diff
@@ -3,53 +3,55 @@
 
 from retrying import retry
 from kafka.consumer.group import KafkaConsumer
+from quintoandar_kafka import IdempotenceClient
+from functools import partial
 
 
-class KafkaConsumerClient:
-
-    def __init__(self, group_id, bootstrap_servers, topic,
-                 processor, deserializer=None, idempotence_client=None):
+class KafkaDefaultConsumer():
+    def __init__(self, processor, **kwargs):
         self.log = logging.getLogger(__name__)
-        self.topic = topic
         self.processor = processor
-        self.deserializer = deserializer or self.default_deserializer
-        self.idempotence_client = idempotence_client or\
-            DefaultIdempotenceClient()
-        self.consumer = self.connect(group_id, bootstrap_servers)
-        self.consumer.subscribe(topic)
+        if kwargs.get('deserializer') is None:
+            kwargs['deserializer'] = self.default_deserializer
+        if kwargs.get('auto_offset_reset') is None:
+            kwargs['auto_offset_reset'] = 'latest'
+        self.consumer = self.connect(kwargs)
+        self.consumer.subscribe(kwargs['topic'])
 
     @retry(stop_max_attempt_number=10, wait_fixed=3000)
-    def connect(self, group_id, bootstrap_servers):
-        return KafkaConsumer(
-            value_deserializer=self.deserializer,
-            group_id=group_id,
-            bootstrap_servers=bootstrap_servers,
-            auto_offset_reset='latest')
+    def connect(self, kwargs):
+        return KafkaConsumer(kwargs)
+
+    def default_deserializer(self, m):
+        try:
+            return json.loads(m.decode('utf8'))
+        except Exception as ex:
+            self.log.error("Failed to decode message: %s", ex, exc_info=True)
+            return {}
 
     def start(self):
         for message in self.consumer:
             if not message.value:
                 continue
 
-            if not self.idempotence_client.is_unique(self.topic, message):
-                continue
-
             self.processor(message)
 
-            self.idempotence_client.mark_consumed_message(self.topic, message)
 
-    def default_deserializer(self, m):
-        try:
-            return json.loads(m.decode('utf8'))
-        except Exception as ex:
-            self.log.error("Failed to decode message: %s", ex, exc_info=True)
-            return {}
+class KafkaIdempotentConsumer(KafkaDefaultConsumer):
 
+    def __init__(self, redis_host, redis_port, processor,
+                 key_extractor, **kwargs):
+        processor = partial(self.idempotent_processor, processor)
+        super().__init__(processor, **kwargs)
+        self.topic = kwargs['topic']
+        self.idempotence_client = IdempotenceClient(
```

Comment:
Ok I don't think we need an IdempotenceClient at all, just put all the methods inside KafkaIdempotentConsumer. _OR_ we keep IdempotenceClient but as a private class declared within KafkaIdempotentConsumer (although that's probably overkill, Python is not Java, we don't need classes for everything...).

As we discussed the client of this lib has no need for a idempotence client, if he needs customization he just passes the processor/key_extractor as parameters to `KafkaIdempotentConsumer`.

The rule here is to expose only what's absolutely necessary for the client who consumes the lib. The rest is clutter. So we can get rid of `quintoandar_kafka.idempotence_client`